### PR TITLE
[Backport] Custom Function Node and Subgraph Output Node refactor

### PR DESCRIPTION
**Pull request by Kink3d**
### Purpose of this PR
Backport of #3029

### CODE REVIEW

File: com.unity.shadergraph/CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,16 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [5.6.0] - 2019-XX-XX
+### Added
+- Added Custom Function node. With this node you can define a custom HLSL function either via string directly in the graph, or via a path to an HLSL include file.
```
Comment (sharlenet) [inline com.unity.shadergraph/CHANGELOG.md]: Added Custom Function node. Use this node to define a custom HLSL function either via string directly in the graph, or via a path to an HLSL include file.

File: com.unity.shadergraph/CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,16 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
...
+
+### Changed
+- Subgraph Output node is now called Outputs.
+- Subgraph Output node now supports renaming of ports.
+- Subgraph Output node now supports all port types.
+- Subgraph Output node now supports reordering ports.
+- When converting nodes to a Sub-Graph, the resulting properties and output ports generated in the Sub-Graph are named for their types by default.
```
Comment (sharlenet) [inline com.unity.shadergraph/CHANGELOG.md]: I'm having trouble understanding what "named for their types" means. Does this mean they are named based on their types?

File: com.unity.shadergraph/CHANGELOG.md (hunk 1 above)
Comment (sharlenet) [reply com.unity.shadergraph/CHANGELOG.md]: Change to:

Added Custom Function node. Use this node to define a custom HLSL function either via string directly in the graph, or via a path to an HLSL file.

File: com.unity.shadergraph/CHANGELOG.md (hunk 2 above)
Comment (sharlenet) [reply com.unity.shadergraph/CHANGELOG.md]: Change to:

When you convert nodes to a Sub-Graph, the resulting properties and output ports, which are generated in the Sub-Graph, are named based on their types by default.

**Comment by zadi0648** [comment]
Concern: plugging a matrix that is bigger than the subgraph output matrix type results in an unhelpful error "Error found during node validation". This does not reproduce / is applicable to other nodes, it is expected that the matrix gets truncated into the smaller type (for example multiplication of nodes of two different sizes results in a mult of the smaller matrix size).

MatrixConcern

**Pull request merged**